pkg/remediator/watch/manager_test.go

```diff
@@ -35,14 +35,15 @@ import (
 	"sigs.k8s.io/cli-utils/pkg/testutil"
 )
 
-func fakeRunnable() Runnable {
+func fakeRunnable(commit string) Runnable {
```

Comment:
What if instead of re-constructing the watch, we just have the commit as a locked field that we can update on the Runnable after it's been started, and then only read it when there's an error? Sounds less clean, but more efficient in terms of starting and stopping watches.

Reply:
The other option is a locked error handler callback.

Reply:
I didn't add any functionality to start/stop watches which wasn't already there before (also this line being commented on is just a fake unit test constructor in case that was unclear).